client/src/views/PasswordReset.jsx

```diff
@@ -1,9 +1,214 @@
+      return () => clearInterval(interval);
+    } else {
+      // console.log("No Images to cycle")
+    }
+  }, [bgImageIdx, imgArr]);
+
+  const handleReset = (e) => {
+    e.preventDefault();
+    setEmailError(false);
+
+    if (!email || !/^[A-Z0-9._%+-]+@[A-Z0-9.-]+\.[A-Z]{2,}$/i.test(email)) {
+      setEmailError(true);
+      return;
+    }
+    console.log(sendPasswordResetEmail(auth, email));
+    sendPasswordResetEmail(auth, email)
+      .then(() => {
+        console.log(sendPasswordResetEmail(auth, email));
+        alert("Check your email");
+      })
+      .catch((error) => {
+        console.error("Password Reset Error:", error);
+        alert("An error occurred. Please try again later.");
+      });
+  };
+
+  useEffect(() => {
+    currentUser ? navigate("/admin") : setShouldLoad(true);
+  }, [currentUser]);
+
+  if (shouldLoad) {
+    return (
+      <Grid container component="main" sx={{ height: "100vh" }}>
+        <CssBaseline />
+        <Grid
+          item
+          xs={false}
+          sm={4}
+          md={7}
+          sx={{
+            backgroundImage: `url(${bgImgString})`,
+            backgroundRepeat: "no-repeat",
+            backgroundSize: "cover",
+            backgroundPosition: "center",
+          }}
+        />
+        <Grid
+          item
+          xs={12}
+          sm={8}
+          md={5}
+          component={Paper}
+          elevation={6}
+          square
+          sx={{
+            background:
+              theme.palette.mode === "dark"
+                ? colors.primary[500]
+                : colors.grey[1000],
+          }}
+        >
+          <Box
+            sx={{
+              my: 8,
+              mx: 4,
+              display: "flex",
+              flexDirection: "column",
+              alignItems: "center",
+            }}
+          >
+            <LogoSVG
+              width={75}
+              height={75}
+              color={colors.blueAccent[500]}
+              sx={{ m: 1 }}
+            />
+            <Typography component="h1" variant="h4">
+              Reset Password
+            </Typography>
+            <Box component="form" noValidate onSubmit={handleReset}>
+              <TextField
+                margin="normal"
+                required
+                fullWidth
+                id="email"
+                label="Email Address"
+                name="email"
+                autoComplete="email"
+                autoFocus
+                value={email}
+                InputLabelProps={{ ...colorTheme.inputLabelProps }}
+                InputProps={{ ...colorTheme.inputProps }}
+                sx={{ ...colorTheme.inputStyling }}
+                onChange={(e) => setEmail(e.target.value)}
+                error={emailError}
+                helperText={emailError ? "Invalid email address" : ""}
+              />
+              <Button
+                type="submit"
```

Comment:
Does MUI use submit buttons?  Normally react takes an onClick event listener, however I don't know the context of submit within MUI.  Please double check and confirm that this is the proper implementation.

Reply:
i'll look more into it